### ISSUE TITLE
PerlTidy never completes tidying some files

### ISSUE THREAD

**Issue by UniqueUsr** [Camelcade/Perl5-IDEA#1872]
# CLICK PREVIEW TAB ^^
1. This repository is for IDE part of perl plugin. Perl-side debugger issues should be posted to the [debugger tracker](https://github.com/hurricup/Perl5-IDEA-Debugger/issues).
2. Before reporting a feature request, please check out our [Wiki](https://github.com/hurricup/Perl5-IDEA/wiki)
3. Before reporting a bug, please, check [known issues](https://github.com/hurricup/Perl5-IDEA/wiki/Known-issues) and [syntax nuances](https://github.com/hurricup/Perl5-IDEA/wiki/Perl-syntax-nuances) sections.
4. Try to search in issues, may be someone already reported your one and you may clarify something or provide more information.
5. When reporting a bug, please, provide as more information as you can. Working example reproducing an error may really speed things up.
6. Make sure that problem still occurs with actual version of the IDE. (see [JetBrains site](https://www.jetbrains.com/))
7. Make sure that problem still occurs with actual version of the Perl plugin. (see [Plugins Repository](https://plugins.jetbrains.com/plugin/7796))
8. At least provide following information:
  * IDE name and version, including build, (e.g. `#IU-143.2370`, available in `Help: About` menu)
  * Perl5 plugin version (available in `File: Settings: Plugins` menu)
  * JRE version (e.g. `Oracle 1.8.0_91-b14`, see `java --version`)
  * Your OS name and version
9. Before posting an issue, please, remove this text. Thank you!

**Comment by hurricup** [comment]
Could you please zip and attach your IDE logs

**Comment by UniqueUsr** [comment]
Logs are attached.

[logs-20181213-095637.zip](https://github.com/Camelcade/Perl5-IDEA/files/2676636/logs-20181213-095637.zip)

**Comment by hurricup** [comment]
1. Could you please try if it works without your custom `-pro` option (and others if there are any)
2. Could you please try if it wokrs from the cmd:
`C:/strawberry/perl/bin\perl.exe "-IC:/JMTF/Release 8/8.8/8.8.0/lib" -IC:/JMTF/Trunk/misc -IC:/JMTF/Trunk/bin -IC:/JMTF/Trunk/utils "-IC:/JMTF/Release 8/8.7/8.7.0/lib" "-IC:/JMTF/Release 8/8.7/8.7.0/misc" "-IC:/JMTF/Release 8/8.7/8.7.0/utils" "-IC:/JMTF/Release 8/8.8/8.8.0/misc" "-IC:/JMTF/Release 8/8.8/8.8.0/utils" "-IC:/JMTF/Release 8/8.8/8.8.0/bin" "-IC:/JMTF/Release 8/8.7/8.7.0/bin" -IC:/JMTF/Trunk/lib C:/strawberry/perl/site/bin/perltidy.bat -st -se -pro=C:\JMTF\Trunk\misc\perltidyrc <fialingfile >res.txt`

**Comment by UniqueUsr** [comment]
Thanks for following up.
1. Does not behave differently with the custom -pro settings.
![image](https://user-images.githubusercontent.com/38534091/50014371-c5e10780-ff91-11e8-9667-ca2bc9a4669f.png)
The thread continues to run in the background.
![image](https://user-images.githubusercontent.com/38534091/50014434-f7f26980-ff91-11e8-8899-4fb1e774f75a.png)
2. The perltidy command runs properly when executed from the cmd prompt.  The res.txt file is attached.
[res.txt](https://github.com/Camelcade/Perl5-IDEA/files/2680904/res.txt)

I also tried messing up the indentation in a file that does not perltidy to see if it still formats but fails to close the thread but it does not - The formatting is still bad and the thread runs forever.

**Comment by hurricup** [comment]
Haven't you figured out the reason? I can't reproduce this :(

**Comment by UniqueUsr** [comment]
No, but I'll look into it more if I get some time.  I wonder if it's McAfee enterprise getting in the way. I've had to add exceptions for other dev tools in the past.




Sent from my mobile device







On Tue, Feb 12, 2019 at 1:05 PM -0500, "Alexandr Evstigneev" <notifications@github.com> wrote:












Haven't you figured out the reason? I can't reproduce this :(

**Comment by UniqueUsr** [comment]
I've spent a bit of time playing with this today and adding exceptions to McAfee had no affect.  When perltidy hangs on a file the task manager shows the Console Window Host and Perl Interpreter as active until I exit the IDE.  On a good perltidy those processes close almost instantly.
![image](https://user-images.githubusercontent.com/38534091/52811403-024ce480-3063-11e9-9d08-565c29b987b7.png)
